Sources/Apollo/ApolloClient.swift

```diff
@@ -110,14 +110,18 @@ extension ApolloClient: ApolloClientProtocol {
 
   @discardableResult
   public func perform<Mutation: GraphQLMutation>(mutation: Mutation,
+                                                 publishResultToStore: Bool = true,
                                                  queue: DispatchQueue = .main,
                                                  resultHandler: GraphQLResultHandler<Mutation.Data>? = nil) -> Cancellable {
-    return self.networkTransport.send(operation: mutation,
-                                      cachePolicy: .default,
-                                      contextIdentifier: nil,
-                                      callbackQueue: queue) { result in
-       resultHandler?(result)
-    }
+    return self.networkTransport.send(
+      operation: mutation,
+      cachePolicy: publishResultToStore ? .fetchIgnoringCacheData : .fetchIgnoringCacheCompletely,
```

Comment:
@designatednerd I went back and forth on this, as I can see both `.returnCacheAndFetch` and `.fetchIgnoringCacheData` being expected behaviors by "default".

I'm willing to revert this to `.default`, but I wanted your take on this.

Reply:
I would say `.default` since for mutations this doesn't actually hit the cache on the way out.

Reply:
Oh right, I glossed over that part and didn't piece it together in my head.

Though, that still doesn't seem right to rely on the interceptor chain setup given that devs might be using their own caching implementation rather than the LegacyCacheRead that Apollo provides.

Then _again_... that's up to them to properly handle everything that they want to do anyway...

I'll just revert this portion to use `.default` like we agreed 😅

Reply:
Ahhh fair point - I think at some point I'd had the `load` on `ApolloStore` set up to throw an error if you tried to call it on anything other than a query (it was previously restricted to things that conformed to the `GraphQLQuery` protocol but that wound up being unworkable with the current setup), but it looks like that got moved up to the intereceptor. I'll chat with @martijnwalraven on that one.

Reply:
Yeah, `cachePolicy` was originally meant to only apply to queries, because the semantics don't really make sense for mutations (or subscriptions for that matter). I think we'll want to revisit cache policies once we have a better idea about the interaction between the request chain and caching, maybe that will allow us to differentiate between operation types again. That would also be the time to think about more flexible error handling, taking TTL into account, etc. So for now, I don't think what we do here really matters, as long as it means the default chain respects `publishResultToStore`.